assets/js/src/shortcake-bakery-admin.js

```diff
@@ -110,4 +110,32 @@ jQuery( document ).ready( function ( $ ) {
 
 			wp.media.editor.open( editor, options );
 		});
+
 });
+
+/**
+ * Attach JS hooks for shortcodes that need them.
+ *
+ */
+if ( typeof wp !== 'undefined' &&
+	 typeof wp.shortcake !== 'undefined' &&
+	 typeof wp.shortcake.hooks !== 'undefined' ) {
+
+	/* Optional fields for GoogleDocs shortcode, displayed conditionally depending on the "type" field */
+	var gdocType = [ ShortcakeBakery.shortcodes.GoogleDocs, 'type' ].join('.');
+	var gdocFields = {
+		all:          [ 'headers', 'start', 'loop', 'delayms', 'allowfullscreen' ],
+		spreadsheet:  [ 'headers' ],
+		presentation: [ 'start', 'loop', 'delayms', 'allowfullscreen' ]
```

Comment:
Yeah, we've talked about that. Would it be good to work that into Shortcake itself? Or should we build out the API here first to validate it before moving it into Shortcake?